java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -170,7 +170,10 @@ class DatabaseImpl(
 
     override suspend fun removeClient(identifier: Int) = clientMutex.withLock {
         clients.remove(nextClientId)
-        // TODO: When all clients are done with the database, close the connection.
+        // When all clients are done with the database, close the connection.
+        if (clients.isEmpty()) {
+            super.close()
```

Comment:
Should we also need to drop the entry in `AndroidSqliteDatabaseManager.dbCache`?
Adding an entry to the dbCache will open the database, if we close the database here w/o deleting it from the dbCache, it seems the subsequent accesses to the database will raise exceptions at once.

Reply:
Yes, I've been staging the PRs into small pieces, not just for easy review, but to be able to tell exactly which commit causes any flakyness if it occurs. :(

Reply:
Gotcha, sgtm.